demos/common/tests/jws_using_mbedtls_ut.c

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#include <string.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/entropy.h"
+
+#include "jws.h"
+#include "demo_config.h"
+#include "FreeRTOSConfig.h"
+
+static mbedtls_entropy_context xEntropyContext;
+static mbedtls_ctr_drbg_context xCtrDrgbContext;
+
+static const char ucManifest[] = "{\"manifestVersion\":\"4\",\"updateId\":{\"provider\":\"ESPRESSIF\",\"name\":\"ESP32-Azure-IoT-Kit\",\"version\":\"1.1\"},\"compatibility\":[{\"deviceManufacturer\":\"ESPRESSIF\",\"deviceModel\":\"ESP32-Azure-IoT-Kit\"}],\"instructions\":{\"steps\":[{\"handler\":\"microsoft/swupdate:1\",\"files\":[\"fc3558477982e3235\"],\"handlerProperties\":{\"installedCriteria\":\"1.0\"}}]},\"files\":{\"fc3558477982e3235\":{\"fileName\":\"azure_iot_freertos_esp32.bin\",\"sizeInBytes\":866128,\"hashes\":{\"sha256\":\"exKJAqfEo69Ok6C6SWy9+Hhp051JbRsXsnMjGSbbJ6o=\"}}},\"createdDateTime\":\"2022-06-03T00:20:33.8421122Z\"}";
```

Comment:
Per offline discussion, we need more test cases here.  An obvious one is the SHA not matching but even things for instance like the token `sjwk` being malformatted in various ways in terms of how many dots are in it on the many cases of that.

Reply:
added a couple more negative tests